OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -658,58 +603,61 @@ class AndroidICCAnalysis(val project: SomeProject) extends FPCFAnalysis {
         m:        Method
     ): Unit = {
         val p = c.params
-        println(c.descriptor)
-        p.size match {
-            case 0 ⇒ //Intent()
+
+        c.descriptor match {
+            case MethodDescriptor(types, _) if types.isEmpty ⇒ //Intent()
                 val intent = new ImplicitIntent(m)
                 evaluateUseSites(tacCode, useSites, intent)
-            case 1 ⇒ //Intent(String action)
+
+            case MethodDescriptor(types, _) if types == FieldTypes(FieldType("Ljava/lang/String;")) ⇒
```

Comment:
Do not create types on the fly. Use, e.g., ObjectType.String for predefined types. Define other types (like the Intent) once on the class level. Also, do not use FieldType("L[...];"), but ObjectType("[...]") directly.